File: CSharp/Patterns/GradientPattern.cs
using System;

namespace CSharp
{
    public class GradientPattern : RTPattern
    {
        public GradientPattern(Color a, Color b) : base(a, b)
        {
            // Nothing to do
        }

        public override Color PatternAt(Point p)
        {
            return a + (b - a) * (p.x - Math.Floor(p.x));
        }

        public override Color UVPatternAt(double u, double v, CubeFace face = CubeFace.None)
        {
            return Color.Black;
        }
    }
}

File: Java/src/main/java/Patterns/GradientPattern.java
package RTC.Patterns;

import RTC.Math.*;
import RTC.Material.*;

public class GradientPattern extends Pattern {
    private Color a, b;

    public GradientPattern(Color A, Color B) {
        super();

        a = A;
        b = B;
    }

    public Color patternAt(Point point) {
        Color distance = b.sub(a);
        double fraction = point.getX() - Math.floor(point.getX());

        return a.add(distance.mul(fraction));
    }

}
